keel-core/src/main/kotlin/com/netflix/spinnaker/keel/lifecycle/LifecycleEventRepository.kt

```diff
@@ -0,0 +1,21 @@
+package com.netflix.spinnaker.keel.lifecycle
+
+import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
+
+interface LifecycleEventRepository {
```

Comment:
I'm not 100% clear why we have this repository as well as `LifecycleMonitorRepository`

Reply:
This is a history of events that have happened. The `LifecycleMonitorRepository` is a working list of things to check.

I thought about doing both in the same repository (monitoring and recording events) but I felt like it was much more confusing. I'd be happy to talk about it with you more, and share what it looks like with actual data.

Reply:
I guess I'm wondering if we need to persist both. I can see the need for the `LifecycleMonitorRepository` but do we actually need to persist the events? They just represent state changes in the task, right?

Reply:
I think that the `LifecycleMonitorRepository` can be replaced in favor of the `task_tracking` table.

Reply:
I don't think so - because (1) I want to monitor tasks a bit differently (read one at a time instead of all of them) and (2) because each monitor knows what it's monitoring and how to respond to the changes. 

When I add a build monitor, it will monitor jenkins and emit events based on that. So it will have a different implementation.